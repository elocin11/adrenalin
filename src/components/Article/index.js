import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Article = React.memo(({ title_long, tag, thumb, slug }) => (
  <article className="article">
    <img
      src={require("./../../assets/images/" + thumb)}
      alt="title"
      className="article-thumb"
    />
    <span className="article-tag">{tag}</span>
    <h4 className="article-title">{title_long}</h4>
    <Link to={`/${slug}`} className="article-link">
      View case study
    </Link>
  </article>
));

Article.propTypes = {
  id: PropTypes.number,
  title_long: PropTypes.string,
  tag: PropTypes.string,
  thumb: PropTypes.string,
  slug: PropTypes.string,
};

export default Article;
