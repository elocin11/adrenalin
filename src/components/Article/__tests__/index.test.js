import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Article from "./../index";

test("Render article with data correctly", () => {
  const data = {
    title_long:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ut libero id auctor.",
    tag: "Toga Group",
    thumb: "1001_thumb.jpg",
    slug: "lorem-ipseum",
  };

  const { getByTestId } = render(
    <BrowserRouter>
      <Article
        title_long={data.title_long}
        tag={data.tag}
        thumb={data.thumb}
        slug={data.slug}
      />
    </BrowserRouter>
  );

  expect(getByTestId("article")).toMatchSnapshot();
});
