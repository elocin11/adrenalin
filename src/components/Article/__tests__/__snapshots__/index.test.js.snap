// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render article with data correctly 1`] = `
<article
  class="article"
  data-testid="article"
>
  <img
    alt="title"
    class="article-thumb"
    src="1001_thumb.jpg"
  />
  <span
    class="article-tag"
  >
    Toga Group
  </span>
  <h4
    class="article-title"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ut libero id auctor.
  </h4>
  <a
    class="article-link"
    href="/lorem-ipseum"
  >
    View case study
  </a>
</article>
`;
